twoliter/src/project/mod.rs

```diff
@@ -215,12 +270,101 @@ impl Project {
     }
 }
 
-/// This represents a container registry vendor that is used in resolving the kits and also
-/// now the bottlerocket sdk
-#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, Ord, PartialOrd)]
-#[serde(rename_all = "kebab-case")]
-pub(crate) struct Vendor {
-    pub registry: String,
+impl Project<SDKLocked> {
+    pub(crate) fn sdk_image(&self) -> ProjectImage {
+        let SDKLocked(lock) = &self.lock;
+        self.as_project_image(&lock.0)
+            .expect("Could not find SDK vendor despite lock resolution succeeding?")
+    }
+}
+
+impl Project<Locked> {
+    /// Fetches all external kits defined in a Twoliter.lock to the build directory
+    pub(crate) async fn fetch(&self, arch: &str) -> Result<()> {
+        let Locked(lock) = &self.lock;
+        lock.fetch(self, arch).await
+    }
+
+    #[expect(dead_code)]
+    pub(crate) fn kits(&self) -> Vec<ProjectImage> {
+        let Locked(lock) = &self.lock;
+        lock.kit
+            .iter()
+            .map(|kit| self.as_project_image(kit))
+            .collect::<Result<_>>()
+            .expect("Could not find kit vendor despite lock resolution succeeding?")
+    }
+
+    pub(crate) fn sdk_image(&self) -> ProjectImage {
+        let Locked(lock) = &self.lock;
+        self.as_project_image(&lock.sdk)
+            .expect("Could not find kit vendor despite lock resolution succeeding?")
+    }
```

Comment:
Is this the right error message?
```suggestion
    pub(crate) fn sdk_image(&self) -> ProjectImage {
        let Locked(lock) = &self.lock;
        self.as_project_image(&lock.sdk)
            .expect("Could not find SDK vendor despite lock resolution succeeding?")
    }
```